src/components/marker-chart/Canvas.js

```diff
@@ -220,12 +221,15 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
             w = 10;
           }
 
+          x = Math.round(x);
+
           const markerIndex = markerTiming.index[i];
           const isHovered = hoveredItem === markerIndex;
           const text = markerTiming.label[i];
           if (isHovered) {
             hoveredElement = { x, y, w, h, uncutWidth, text };
-          } else {
+          } else if (x !== previousDrawnPixel) {
```

Comment:
not: Not checking the duration here (on the contrary of the check below), because I rely on the fact that the MarkerTiming computation put longer markers on different lines usually.

Reply:
I believe this assumption is not correct. I'm getting some marker flashing:

STR:
 * [View this profile](https://deploy-preview-1722--perf-html.netlify.com/public/dbc6d5633d0a6894add3ab868bb439f179dcf7b4/marker-chart/?globalTrackOrder=0-1-2-3-4&hiddenGlobalTracks=1-2&range=6.1896_8.8912&thread=5&v=3)
 * Shift scroll to zoom the chart.
 * The second row of the DOMEvent `react-invokeguardedcallback` flashes in and out.

> <img width="286" alt="image" src="https://user-images.githubusercontent.com/1588648/52220862-7f0ee000-2865-11e9-8393-21993e50deb7.png">

> <img width="264" alt="image" src="https://user-images.githubusercontent.com/1588648/52220873-8635ee00-2865-11e9-8be1-ec3cf3ca2de1.png">

Reply:
Ah thanks, I thought it could be on the edge, and now you proved it's on the wrong side of the edge. Will definitely fix!

Reply:
The marker timing mechanism uses `>=` which is the cause for this problem; this also means all dot markers with the same name happening at the same exact time will be drawn exactly at the same space. This may be a problem...

I won't change this and rather change my condition here, but we should keep it in mind.